switchy/apps/__init__.py

```diff
@@ -1,6 +1,90 @@
+# vim: tabstop=4 softtabstop=4 shiftwidth=4 textwidth=80 smarttab expandtab
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 '''
 Built-in applications
 '''
+
+# Registry of switchy apps
+_apps = {}
+
+def app(*args, **kwargs):
+    '''Decorator to register switchy application classes
+       You can call it as:
+
+       @app
+       class CoolAppController(object):
+           pass
+
+       This will register the class as a switchy app
+       The name of the app defaults to the class name
+       The help for the app defaults to the class doc string
+
+       But you can also provide alternative help and name
+       decorator arguments to register your app with:
+
+       @app('Just try it!', 'CoolName')
+       class CoolAppController(object):
+           pass
+
+       or with keyword args:
+
+       @app(name='CoolName', help='Some help text for the user')
+       class CoolAppController(object):
+           pass
+    '''
+    name = None
+    help = None
+    if len(args) >= 1:
+        if type(args[0]) is type:
+            return register(args[0], None, None)
+        help = args[0]
+        if len(args) > 1:
+            name = args[1]
+    if 'help' in kwargs:
+        help = kwargs['help']
+    if 'name' in kwargs:
+        name = kwargs['name']
+
+    def inner(cls):
+        return register(cls, name, help)
+    return inner
+
+def register(cls, name=None, help=None):
+    if not name:
+        name = cls.__name__
+    if not help:
+        help = cls.__doc__
+        if not help:
+            help = '(No help for app {})'.format(name)
+
+    if name in _apps:
+        return cls
+
+    from collections import namedtuple
+    cls._switchy_app = namedtuple('SwitchyApp', ['name', 'help', 'cls'])
+    cls._switchy_app.name = name
+    cls._switchy_app.help = help
+    cls._switchy_app.cls = cls
+
+    _apps[name] = cls
+    return cls
+
+def next():
```

Comment:
This isn't semantically consistent with python iterables. Normally a generator like this would have name like `iterapps` whereas `next` usually only delivers the 'next' single element from an iterable.